pkg/operator/controllers/banner/banner.go

```diff
@@ -0,0 +1,56 @@
+package banner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+func (r *Reconciler) reconcileBanner(ctx context.Context, instance *arov1alpha1.Cluster) (err error) {
+	switch instance.Spec.Banner.Content {
+	case arov1alpha1.BannerDisabled:
+		err = r.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, BannerName, metav1.DeleteOptions{})
+		if err != nil && kerrors.IsNotFound(err) {
+			// we don't care if the object doesn't exist
+			err = nil
+		}
+		return err
+	case arov1alpha1.BannerContactSupport:
+		_, err := r.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, BannerName, metav1.GetOptions{})
+		if err != nil && kerrors.IsNotFound(err) {
+			// if the object doesn't exist Create
+			_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, r.newBanner(TextContactSupport, instance.Spec.ResourceID), metav1.CreateOptions{})
+			return err
+		}
+		if err != nil {
+			// if there's a different error abort
+			return err
+		}
+		// if there's no errors, object found then update
+		_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, r.newBanner(TextContactSupport, instance.Spec.ResourceID), metav1.UpdateOptions{})
+		return err
+	}
+	return fmt.Errorf("wrong banner setting '%s'", instance.Spec.Banner.Content)
```

Comment:
All this switch would be bit more readable like this:
```
switch instance.Spec.Banner.Content {
	case arov1alpha1.BannerDisabled:
		err = r.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, BannerName, metav1.DeleteOptions{})
		if err != nil && kerrors.IsNotFound(err) {
			// we don't care if the object doesn't exist
			return nil
		}
		return err
	case arov1alpha1.BannerContactSupport:
		_, err := r.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, BannerName, metav1.GetOptions{})
		if !kerrors.IsNotFound(err) {
			return err
		}
		
		// create or update
		if kerrors.IsNotFound(err) {
			// if the object doesn't exist Create
			_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, r.newBanner(TextContactSupport, instance.Spec.ResourceID), metav1.CreateOptions{})
			return err
		}


		// if there's no errors, object found then update
		_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, r.newBanner(TextContactSupport, instance.Spec.ResourceID), metav1.UpdateOptions{})
		return err
	}
```
Don't yuo think?

Reply:
Early escape looks cleaner since we surface all actual error states early (return err) and then take care of other err values.
Modified the switch.